### PR TITLE
RATIS-1862 Add the parameter whether to take Snapshot when stopping to adapt to different services

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira1862](https://issues.apache.org/jira/browse/RATIS-1862)

### CODE REVIEW

**Comment by OneSizeFitsQuorum** [comment]
@szetszwo PTAL

File: ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java [hunk 1]
```diff
@@ -586,6 +586,18 @@ static void setAutoTriggerEnabled(RaftProperties properties, boolean autoTrigger
       setBoolean(properties::setBoolean, AUTO_TRIGGER_ENABLED_KEY, autoTriggerEnabled);
     }
 
+    /** whether trigger snapshot when stop raft server */
+    String TRIGGER_SNAPSHOT_WHEN_STOP_ENABLED_KEY = PREFIX + ".trigger.snapshot.when.stop.enabled";
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java:590]: Let's rename it to `PREFIX + ".trigger-when-stop.enabled"`.

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks a lot for working on this!  Just have a comment on the new conf name.  The change looks good other than that.

File: ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java]: It took me a long time to come up with this name, and now thank you for the better name!

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**